ctk/common/src/test/kotlin/org/codice/compilance/verification/profile/SingleSignOnProfileVerifierSpec.kt

```diff
@@ -0,0 +1,123 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.compilance.verification.profile
+
+import io.kotlintest.matchers.string.shouldContain
+import io.kotlintest.shouldThrow
+import io.kotlintest.specs.StringSpec
+import org.codice.compliance.Common
+import org.codice.compliance.SAMLComplianceException
+import org.codice.compliance.SAMLProfiles_3_1_a
+import org.codice.compliance.SAMLProfiles_3_1_c
+import org.codice.compliance.utils.HOLDER_OF_KEY_URI
+import org.codice.compliance.utils.PERSISTENT_ID
+import org.codice.compliance.utils.SUCCESS
+import org.codice.compliance.verification.profile.SingleSignOnProfileVerifier
+import java.time.Instant
+import java.util.UUID
+
+@Suppress("StringLiteralDuplication")
+class SingleSignOnProfileVerifierSpec : StringSpec() {
+    private val correctType = "xsi:type=\"s2:KeyInfoConfirmationDataType\""
+    private val correctKeyInfo = """
+                |<ds:KeyInfo>
+                |  <ds:KeyName>key</ds:KeyName>
+                |</ds:KeyInfo>
+                """.trimMargin()
+
+    init {
+        "response with correct holder-of-key subject confirmation should pass" {
+            Common.buildDom(createResponse(createSubjConfData(keyInfo = correctKeyInfo))).let {
+                SingleSignOnProfileVerifier(it).verifyHolderOfKey()
+            }
+        }
+
+        "response with no subject confirmation data type should pass" {
+            Common.buildDom(createResponse(createSubjConfData("", correctKeyInfo))).let {
+                SingleSignOnProfileVerifier(it).verifyHolderOfKey()
+            }
+        }
+
+        "response with no subject confirmation data should fail" {
+            Common.buildDom(createResponse("")).let {
+                shouldThrow<SAMLComplianceException> {
+                    SingleSignOnProfileVerifier(it).verifyHolderOfKey()
+                }.message?.shouldContain(SAMLProfiles_3_1_a.message)
+            }
+        }
+
+        "response with no key info should fail" {
+            Common.buildDom(createResponse(createSubjConfData(keyInfo = ""))).let {
+                shouldThrow<SAMLComplianceException> {
+                    SingleSignOnProfileVerifier(it).verifyHolderOfKey()
+                }.message?.shouldContain(SAMLProfiles_3_1_a.message)
+            }
+        }
+
+        "response with multiple key values should fail" {
+            val incorrectKeyInfo = """
+                |<ds:KeyInfo>
+                |  <ds:KeyValue>value</ds:KeyValue>
+                |  <ds:KeyValue>value</ds:KeyValue>
+                |</ds:KeyInfo>
+                """.trimMargin()
+
+            Common.buildDom(createResponse(createSubjConfData(keyInfo = incorrectKeyInfo))).let {
+                shouldThrow<SAMLComplianceException> {
+                    SingleSignOnProfileVerifier(it).verifyHolderOfKey()
+                }.message?.shouldContain(SAMLProfiles_3_1_c.message)
+            }
+        }
```

Comment:
❓ Should we be testing for when the `type`'s namespace is not equal to the `<Assertion>`'s namespace?